### PR TITLE
[AL-2996] Fix metadata uploads with local file datarow upload

### PR DESCRIPTION
**Pull request by kkim-labelbox**
- Fixed metadata not uploading when data row is a local file
- Added test cases

### CODE REVIEW

File: labelbox/schema/dataset.py [hunk 1]
```diff
@@ -260,11 +260,10 @@ def upload_if_necessary(item):
             row_data = item['row_data']
             if os.path.exists(row_data):
                 item_url = self.client.upload_file(item['row_data'])
```
Comment (jtsodapop) [inline labelbox/schema/dataset.py:262]: **suggestion** can we change this `item_url = self.client.upload_file(item['row_data'])` to this?

`item_url = self.client.upload_file(row_data)`

File: labelbox/schema/dataset.py [hunk 2]
```diff
@@ -260,11 +260,10 @@ def upload_if_necessary(item):
...
-                item = {
-                    "row_data": item_url,
-                    "external_id": item.get('external_id', item['row_data']),
-                    "attachments": item.get('attachments', [])
-                }
+                item['row_data'] = item_url
+                if 'external_id' not in item:
+                    # Default `external_id` to local file name
+                    item['external_id'] = row_data
```
Comment (msokoloff1) [inline labelbox/schema/dataset.py:266]: How is this different?

Comment (kkim-labelbox) [reply labelbox/schema/dataset.py:266]: it uses the original `item`, and alters it, rather than constructing a brand new `item`

Comment (jtsodapop) [inline labelbox/schema/dataset.py:266]: **question** should we include back attachments?

File: labelbox/schema/dataset.py [hunk 3]
```diff
@@ -260,11 +260,10 @@ def upload_if_necessary(item):
             row_data = item['row_data']
             if os.path.exists(row_data):
                 item_url = self.client.upload_file(item['row_data'])
-                item = {
-                    "row_data": item_url,
-                    "external_id": item.get('external_id', item['row_data']),
-                    "attachments": item.get('attachments', [])
-                }
+                item['row_data'] = item_url
+                if 'external_id' not in item:
```
Comment (jtsodapop) [inline labelbox/schema/dataset.py:264]: **question** im a little confused on how this resolved the issue. from the previous code, wouldn't it also do the same?

`"external_id": item.get('external_id', item['row_data']),`

File: labelbox/schema/dataset.py (hunk 2 above)
Comment (kkim-labelbox) [reply labelbox/schema/dataset.py:266]: If attachments are included in the original `item`, they will remain

File: labelbox/schema/dataset.py (hunk 3 above)
Comment (jtsodapop) [reply labelbox/schema/dataset.py:264]: resolving convo as this was asked in another comment

File: labelbox/schema/dataset.py (hunk 2 above)
Comment (jtsodapop) [reply labelbox/schema/dataset.py:266]: resolving as i now understand attachments will already be part of existing item

**Review by jtsodapop — APPROVED** [review]

File: labelbox/schema/dataset.py (hunk 2 above)
Comment (msokoloff1) [reply labelbox/schema/dataset.py:266]: What is the need for this?

Comment (msokoloff1) [reply labelbox/schema/dataset.py:266]: oh sorry, I should have read the description.

**Review by msokoloff1 — APPROVED** [review]

File: labelbox/schema/dataset.py (hunk 2 above)
Comment (kkim-labelbox) [reply labelbox/schema/dataset.py:266]: No worries, it's a subtle change haha

**Pull request merged**